test/notifications/display_test.dart

```diff
@@ -208,6 +211,158 @@ void main() {
+
+      // Ensure that on Android 10, notification sounds aren't being copied to
+      // the media store, and resource file is used directly.
+      await NotificationChannelManager.ensureChannel();
+      check(androidNotificationHost.takeCopySoundResourceToMediaStoreCalls()).length.equals(0);
+
+      final defaultSoundResourceName =
+        NotificationChannelManager.kDefaultNotificationSound.resourceName;
+      final soundUrl =
+        'android.resource://com.zulip.flutter/raw/$defaultSoundResourceName';
+      check(androidNotificationHost.takeCreatedChannels()).single
+        ..id.equals(NotificationChannelManager.kChannelId)
+        ..name.equals('Messages')
+        ..importance.equals(NotificationImportance.high)
+        ..lightsEnabled.equals(true)
+        ..soundUrl.equals(soundUrl)
+        ..vibrationPattern.isNotNull().deepEquals(
+            NotificationChannelManager.kVibrationPattern);
+    });
+
+    test('notification sound resource files are being copied to the media store', () async {
+      addTearDown(testBinding.reset);
+      final androidNotificationHost = testBinding.androidNotificationHost;
+
+      await NotificationChannelManager.ensureChannel();
+      check(androidNotificationHost.takeCopySoundResourceToMediaStoreCalls())
+        .deepEquals(NotificationSound.values.map((e) => (
+          sourceResourceName: e.resourceName,
+          targetFileDisplayName: e.fileDisplayName),
+        ));
+
+      // Ensure the default source URL points to a file in the media store,
+      // rather than a resource file.
+      final defaultSoundResourceName =
+        NotificationChannelManager.kDefaultNotificationSound.resourceName;
+      final soundUrl =
+        androidNotificationHost.fakeStoredNotificationSoundUrl(defaultSoundResourceName);
+      check(androidNotificationHost.takeCreatedChannels()).single
+        ..id.equals(NotificationChannelManager.kChannelId)
+        ..name.equals('Messages')
+        ..importance.equals(NotificationImportance.high)
+        ..lightsEnabled.equals(true)
+        ..soundUrl.equals(soundUrl)
+        ..vibrationPattern.isNotNull().deepEquals(
+            NotificationChannelManager.kVibrationPattern);
+    });
+
+    test('notification sounds are not copied again if they were previously copied', () async {
+      addTearDown(testBinding.reset);
+      final androidNotificationHost = testBinding.androidNotificationHost;
+
+      // Emulate that all notifications sounds are already in the media store.
+      androidNotificationHost.setupStoredNotificationSounds(
+        NotificationSound.values.map((e) => StoredNotificationSound(
+          fileName: e.fileDisplayName,
+          isOwned: true,
+          contentUrl: androidNotificationHost.fakeStoredNotificationSoundUrl(e.resourceName)),
+        ).toList(),
+      );
+
+      await NotificationChannelManager.ensureChannel();
+      check(androidNotificationHost.takeCopySoundResourceToMediaStoreCalls()).length.equals(0);
```

Comment:
nit: line too long, with key information past 80 columns; also can simplify to `isEmpty`:

```suggestion
      check(androidNotificationHost.takeCopySoundResourceToMediaStoreCalls())
        .isEmpty();
```